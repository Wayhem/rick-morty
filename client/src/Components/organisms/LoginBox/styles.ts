/* eslint-disable @typescript-eslint/no-explicit-any */
import styled from 'styled-components'
import OutlinedButton from 'Components/atoms/outlinedButton'
import StandardButton from 'Components/atoms/standardButton'

export const Box = styled.div`
  border: 1px solid ${props => props.theme.colors.gray};
  box-shadow: 2px 2px 5px 5px ${props => props.theme.colors.transpBlack};
  border-radius: 4px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.theme.colors.white};
  max-width: 700px;
  width: 50%;

  @media (max-width: 1024px) {
    width: 70%;
  }

  @media (max-width: 768px) {
    width: 90%;
  }
`

export const InputContainer = styled.div`
  margin: 8px 0;
`

export const SubmitButton = styled(StandardButton)``

export const ClearButton = styled(OutlinedButton)``

export const Row = styled.div<any>`
  display: flex;
  width: 100%;
  justify-content: space-between;
  margin-top: ${props => props.marginTop || '1rem'};
  background-color: ${props => props.backgroundColor};
  border-top: ${props => props.borderTop};
`
export const Form = styled.form`
  padding: 1rem;
  width: 100%;
`

export const FormTypeButton = styled.div<any>`
  padding: 12px;
  border-right: ${props => props.borderRight};
  cursor: pointer;
  text-align: center;
  width: 50%;
  transition: all 0.2s ease-in;
  font-size: 1.1rem;
  color: ${props => props.theme.colors.orange};
  border-top: ${props => props.borderTop};

  &:hover {
    opacity: ${props => props.isSelected ? '1' : '0.5'};
    background-color: ${props => props.isSelected ? props.theme.colors.white : props.theme.colors.gray};
  }

  &:active {
    opacity: 1;
    background-color: ${props => props.theme.colors.white};
  }
`

export const FieldsContainer = styled.div`
  min-height: 136px;
`
