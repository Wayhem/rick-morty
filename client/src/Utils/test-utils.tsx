import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult, Queries } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'
import { MemoryRouter as Router } from 'react-router-dom'
import { store } from 'Store'
import { theme } from 'Config/styled'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AllTheProviders: FC = ({ children }: { children?: any }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  </ThemeProvider>
)

AllTheProviders.defaultProps = {
  children: <></>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult<Queries, HTMLElement> => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'

export { customRender as render }
