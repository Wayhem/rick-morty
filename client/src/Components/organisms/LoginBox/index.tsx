import React, { useState } from 'react'
import { useDispatch } from 'react-redux'

import TextInput from 'Components/atoms/TextInput'
import { Box, InputContainer, SubmitButton, Row, ClearButton } from 'Components/organisms/LoginBox/styles'

import { positionShift } from 'Config/constants/inputLabelShift'

import { createUserAction } from 'Store/actions/authActions'

const LoginBox = () => {
  const [username, setUsername] = useState('')
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')

  const dispatch = useDispatch()

  const clearFields = () => {
    setUsername('')
    setEmail('')
    setPassword('')
  }
  const handleSubmit = (e: any) => {
    e.preventDefault()
    dispatch(createUserAction(username, password, email))
  }

  return (
    <Box>
      <form onSubmit={handleSubmit} autoComplete='off'>
        <InputContainer>
          <TextInput
            label='Email'
            placeholder='Email'
            inputId='emailInput'
            type='email'
            value={email}
            onChange={e => setEmail(e.target.value)}
            labelShift={positionShift.up}
            required
          />
        </InputContainer>
        <InputContainer>
          <TextInput
            label='Username'
            placeholder='Username'
            inputId='usernameInput'
            value={username}
            onChange={e => setUsername(e.target.value)}
            required
          />
        </InputContainer>
        <InputContainer>
          <TextInput
            label='Password'
            placeholder='Password'
            inputId='passwordInput'
            value={password}
            type='password'
            onChange={e => setPassword(e.target.value)}
            labelShift={positionShift.down}
            required
          />
        </InputContainer>
        <Row>
          <ClearButton type='button' onClick={clearFields}>Clear</ClearButton>
          <SubmitButton type='submit'>Submit</SubmitButton>
        </Row>
      </form>
    </Box>
  )
}

export default LoginBox
