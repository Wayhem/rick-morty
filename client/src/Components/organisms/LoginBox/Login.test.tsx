import React from 'react'
import LoginBox from 'Components/organisms/LoginBox'
import { render } from 'Utils/test-utils'

test('LoginBox renders and matches snapshot', () => {
  const { container } = render(<LoginBox />)
  expect(container).toMatchSnapshot()
})
