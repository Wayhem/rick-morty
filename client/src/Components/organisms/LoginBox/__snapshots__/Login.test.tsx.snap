// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginBox renders and matches snapshot 1`] = `
<div>
  <div
    class="sc-eCssSg daGpOG"
  >
    <form
      autocomplete="off"
      class="sc-pFZIQ vYZga"
    >
      <div
        class="sc-kEjbxe SSvhe"
      >
        <div
          class="sc-jSgupP gsBzSk"
        >
          <div
            class="sc-bdfBwQ VfWiw"
          >
            <input
              class="sc-gsTCUz ixbftl"
              id="emailInput"
              placeholder="email"
              required=""
              type="email"
              value=""
            />
            <label
              for="emailInput"
            >
              Email
            </label>
          </div>
        </div>
        <div
          class="sc-jSgupP gsBzSk"
        >
          <div
            class="sc-bdfBwQ VfWiw"
          >
            <input
              class="sc-gsTCUz gPDLux"
              id="usernameInput"
              placeholder="username"
              required=""
              type="text"
              value=""
            />
            <label
              for="usernameInput"
            >
              Username
            </label>
          </div>
        </div>
        <div
          class="sc-jSgupP gsBzSk"
        >
          <div
            class="sc-bdfBwQ VfWiw"
          >
            <input
              class="sc-gsTCUz gPDLux"
              id="passwordInput"
              placeholder="password"
              required=""
              type="password"
              value=""
            />
            <label
              for="passwordInput"
            >
              Password
            </label>
          </div>
        </div>
      </div>
      <div
        class="sc-fubCfw kxlvIP"
      >
        <button
          class="sc-dlfnbm sc-iBPRYJ fwNmzU"
          type="button"
        >
          Clear
        </button>
        <button
          class="sc-hKgILt sc-gKsewC lodOJa"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
    <div
      class="sc-fubCfw PgBbZ"
    >
      <div
        class="sc-jrAGrp gITcIJ"
      >
        Register
      </div>
      <div
        class="sc-jrAGrp ecoQRy"
      >
        Login
      </div>
    </div>
  </div>
</div>
`;
